### ISSUE TITLE
Connect after pairing fails

### ISSUE THREAD

**Issue by rmartinsanta** [MuntashirAkon/libadb-android#4]
After a device has been successfully paired, how do we connect ADB? I may have misunderstood the documentation, but I am doing something like:

```java
AdbConnectionManager connectionManager = AdbConnectionManager.getInstance(context);
boolean paired = connectionManager.pair(ip, PAIR_PORT, PIN);
boolean connected = connectionManager.connect(ip, ADB_PORT); // Fails with ECONNREFUSED, port is closed. Tested same port as pairing port and 5555
```

Thanks in advance.

**Comment by mirfatif** [comment]
Connection port is different than the pairing port. The former is randomly assigned after pairing is successful.
See screenshots [here](https://mirfatif.github.io/WhatsRunning/help/help2#adb-help-11).

**Comment by MuntashirAkon** [comment]
See the official documentation on Wireless debugging: https://developer.android.com/studio/command-line/adb#connect-to-a-device-over-wi-fi-android-11+

TL;DR; As @mirfatif has pointed out, ADB pairing and connection ports are different.

If the app is being run in the same device where ADB daemon is active, you forget about port numbers (11+) by using the auto-connect methods i.e. `autoConnect`, `connectTcp` or `connectTls`.

For ADB pairing, you can retrieve ADB host and port automatically using something like this:

```java
    @WorkerThread
    @NonNull
    public static Pair<String, Integer> getLatestAdbPairingDaemon(@NonNull Context context, long timeout, @NonNull TimeUnit unit)
            throws InterruptedException, IOException {
        AtomicInteger atomicPort = new AtomicInteger(-1);
        AtomicReference<String> atomicHostAddress = new AtomicReference<>(null);
        CountDownLatch resolveHostAndPort = new CountDownLatch(1);

        AdbMdns adbMdnsPairing = new AdbMdns(context, AdbMdns.SERVICE_TYPE_TLS_PAIRING, (hostAddress, port) -> {
            atomicHostAddress.set(hostAddress.getHostAddress());
            atomicPort.set(port);
            resolveHostAndPort.countDown();
        });
        adbMdnsPairing.start();

        try {
            if (!resolveHostAndPort.await(timeout, unit)) {
                throw new InterruptedException("Timed out while trying to find a valid host address and port");
            }
        } finally {
            adbMdnsPairing.stop();
        }

        String host = atomicHostAddress.get();
        int port = atomicPort.get();
        if (host == null || port == -1) {
            throw new IOException("Could not find any valid host address or port");
        }
        return new Pair<>(host, port);
    }
```
